src/core/blockchain/near/client.ts

```diff
@@ -0,0 +1,126 @@
+import { BlockchainGenericClient, IFeeOptions, ChainIdType, IBlockInfo } from '../types';
+import { networks } from './networks';
+import {
+    createAccount,
+    transfer,
+    addKey,
+    fullAccessKey,
+    createTransaction,
+    signTransaction
+} from 'nearlib/src.ts/transaction';
+import { PublicKey, KeyPair, serialize } from 'nearlib/src.ts/utils';
+import BN from 'bn.js';
+import BigNumber from 'bignumber.js';
+import sha256 from 'js-sha256';
+
+export class Client extends BlockchainGenericClient {
+    constructor(chainId: ChainIdType) {
+        super(chainId, networks);
+    }
+
+    public async getBalance(address: string): Promise<BigNumber> {
+        const res = await this.rpc.call('query', [`account/${address}`, '']);
+
+        return new BigNumber(res.result.amount);
+    }
+
+    public async getNonce(address: string, publicKey?: string): Promise<number> {
+        const res = await this.rpc.call('query', [`access_key/${address}/${publicKey}`, '']);
+
+        return res.result.nonce + 1;
+    }
+
+    public async getCurrentBlock(): Promise<IBlockInfo> {
+        const res = await this.rpc.call('status');
+
+        return {
+            number: res?.result?.sync_info?.latest_block_height,
+            hash: res?.result?.sync_info?.latest_block_hash
+        };
+    }
+
+    public async sendTransaction(signedTransaction): Promise<string> {
+        const res = await this.rpc.call('broadcast_tx_commit', [signedTransaction]);
+
+        return res?.result?.transaction?.hash;
+    }
+
+    public async calculateFees(
+        from: string,
+        to: string,
+        amount?,
+        contractAddress?
+    ): Promise<IFeeOptions> {
+        return {
+            gasPrice: '937144500000',
+            gasLimit: '1'
+        };
+    }
+
+    public async checkAccountIdValid(accountId: string): Promise<boolean> {
+        try {
+            const res = await this.rpc.call('query', [`account/${accountId}`, '']);
+
+            if (res.result) {
+                // account id already taken
+                return false;
+            } else {
+                // valid account id
+                return true;
+            }
+        } catch (err) {
+            return false;
+        }
+    }
+
+    public async createAccount(
+        newAccountId: string,
+        publicKey: string,
+        chainId: ChainIdType
+    ): Promise<any> {
+        const SENDER_ACCOUNT_ID = 'tibi';
+        const SENDER_PRIVATE_KEY =
+            'ed25519:47XC2WW9NWmnvpAE48Jjy8qdgrEjHaovXFDGUrFhKnvvD1mv8PAtSav97wroJx5E8fd3Z2zQGZwRA7e3krzQAm49';
```

Comment:
hardcoded?

Reply:
This will be moved to our service client in Firebase.